code/modules/mob/living/carbon/human/inventory.dm

```diff
@@ -187,11 +187,18 @@
 
 //This is an UNSAFE proc. Use mob_can_equip() before calling this one! Or rather use equip_to_slot_if_possible() or advanced_equip_to_slot_if_possible()
 //set redraw_mob to 0 if you don't wish the hud to be updated - if you're doing it manually in your own proc.
-/mob/living/carbon/human/equip_to_slot(obj/item/W as obj, slot, redraw_mob = 1)
+/mob/living/carbon/human/equip_to_slot(obj/item/W as obj, slot, redraw_mob = 1, equip_speed_override = FALSE)
 	if(!slot) return
 	if(!istype(W)) return
 	if(!has_organ_for_slot(slot)) return
 
+	if(W.equip_speed && !equip_speed_override) //check to see if we want a delay and check to see we don't want an override for uses such as admin equipping (cmd_admin_dress). 

```

Comment:
Might be wise to give text feedback at the start of the bar as well.
ie; `to_chat(src, "<span class='notice'>You start to struggle into the [W].</span>")`

Reply:
Would also make sense if the text corresponded to the speed for that suit type- i.e., a mining hardsuit would use the struggle example above, but a syndicate elite suit(instant) would possibly say, "You slip into the syndicate elite hardsuit."